ndc-client/src/models.rs

```diff
@@ -334,13 +334,42 @@ pub enum Aggregate {
 }
 // ANCHOR_END: Aggregate
 
+// ANCHOR: NestedObject
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[schemars(title = "NestedObject")]
+pub struct NestedObject {
+    pub fields: IndexMap<String, Field>,
+}
+// ANCHOR_END: NestedObject
+
+// ANCHOR: NestedArray
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[schemars(title = "NestedArray")]
+pub struct NestedArray {
+    pub fields: Box<Option<NestedField>>,
+}
+// ANCHOR_END: NestedArray
+
+// ANCHOR: NestedField
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[schemars(title = "NestedField")]
+pub enum NestedField {
+    Object(NestedObject),
+    Array(NestedArray)
+}
+// ANCHOR_END: NestedField
+
 // ANCHOR: Field
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
 #[serde(tag = "type", rename_all = "snake_case")]
 #[schemars(title = "Field")]
 pub enum Field {
     Column {
         column: String,
+        fields: Option<NestedField>
```

Comment:
```suggestion
        /// When the type of the column is a (possibly-nullable) array or object,
        /// the caller can request a subset of the complete column data,
        /// by specifying fields to fetch here.
        /// If omitted, the column data will be fetched in full.
        fields: Option<NestedField>
```

We probably ought to add a section to the spec to clarify the behavior for nullable types. E.g. if `person.address` is null and the user requests `{ person { address { city } } }`, then I think we can specify that `{ person: { address : null } }` ought to be returned.

I'd also be curious to bikeshed this name a bit. It's the best name I can currently think of, but it also a bit strange that it's field_s_ plural, but there's only one structure being described. It makes sense, because the leaves of the structure are always `FieldObject`, but that might not be the case forever. In general, it's just going to be a type-directed set of instructions for fetching a subset of data. But I can't think of a better name for that. Perhaps `transform`, but that's confusing for other reasons.